### DISCUSSION TITLE
Joining Multiple Indices’ Correlated Data in One ReactiveBase Component

### DISCUSSION THREAD

**Discussion by samstalner** [appbaseio/reactivesearch#1573] [Q&A] [answered]
Hi there,

I was reading through the documentation, and was curious whether joining multiple indices’ data within a single ReactiveBase component is possible. I am aware that one can add multiple indices to the `app` prop (e.g. `<ReactiveBase url='my_cluster_url' app='index1,index2'>`), but I am interested in augmenting/merging search results such that the correlated information is available for each search result. For example, let’s say I have two indices Orders and Customers:

Orders index:
```
{
        …,
	customer:  {
		id: 123
	},
	orderDate: '12-15-2020',
	item: 'Yellow Shoe'
	…
}
```

Customers index:
```
{
	…,
	_id: 123,
	name: 'John Smith',
	dob: '01-02-1990',
	…
}
```

I now want to use ReactiveList to create a custom component that renders out the `item` and `orderDate` from Orders, as well as the corresponding customer’s `name` and `dob` from Customers (correlated on the `id` field). This would mean having a result object that contains a join of the fields from both the Orders and Customers index for that id.

Is there a way to do this natively using a ReactiveBase component? Are there examples of this somewhere?

Thank you so much!

**Comment by siddharthlatest** [comment, marked as answer]
There isn't a native way to do joins within ReactiveSearch. There are some workarounds though:

1. You can use `transformResponse` prop in ReactiveBase to query the other index and then stitch the data together. There's an example here that's showing that: https://docs.appbase.io/docs/reactivesearch/v3/overview/reactivebase/#props.
2. Another possible option is for you to change the data model within Elasticsearch: if you use the `nested` type to capture the customers within the order (or vice-versa), now you can query this more easily within ReactiveSearch.

**Reply by samstalner** [reply]
Thank you for your response @siddharthlatest! I was experimenting with the first option (using `transformResponse`) and just wanted to verify whether there was a cleaner way about it within ReactiveSearch. We will look into using the second option (creating a new index with all relevant information nested, and using the `nestedField` prop for queries). Thanks again!
